docs: add an example for setting the state outside event (#302) (#303)
pmndrs/zustand: readme.md

const useStore = create(redux(reducer, initialState))
```

## Calling actions outside a React event handler

Because React handles `setState` synchronously if it's called outside an event handler. Updating the state outside an event handler will force react to update the components synchronously, therefore adding the risk of encountering the zombie-child effect.
In order to fix this, the action needs to be wrapped in `unstable_batchedUpdates`

```jsx

> import { unstable_batchedUpdates } from 'react-dom'; // or react-native

Reviewer: ```suggestion
import { unstable_batchedUpdates } from 'react-dom' // or 'react-native'
```

const useStore = create((set) => ({
  fishes: 0,

>   increaseFishes: () => set((prev) => ({ fished: prev.fished + 1 }))

Reviewer: ```suggestion
  increaseFishes: () => set((prev) => ({ fishes: prev.fishes + 1 }))
```

const useStore = create(redux(reducer, initialState))
```

## Calling actions outside a React event handler

Because React handles `setState` synchronously if it's called outside an event handler. Updating the state outside an event handler will force react to update the components synchronously, therefore adding the risk of encountering the zombie-child effect.
In order to fix this, the action needs to be wrapped in `unstable_batchedUpdates`

```jsx

> import create from 'zustand'
> import { unstable_batchedUpdates } from 'react-dom';
> 
> export const useStore = create(persist(
>   (set) => ({
>     fishes: 0,
>     addAFish: () => set({ fish: get().fish + 1 })
>   }),
> ))
> 
> const addAFish = useStore((state) => state.addAFish)
> 
> async function onClickHandler() {
>   await asyncFunc()
>   unstable_batchedUpdates(() => {
>     addAFish()
>   })
> }

Reviewer: ```suggestion
import { unstable_batchedUpdates } from 'react-dom'; // or react-native

const useStore = create((set) => ({
  fishes: 0,
  increaseFishes: () => set((prev) => ({ fished: prev.fished + 1 }))
}))

const nonReactCallback = () => {
  unstable_batchedUpdates(() => {
    useStore.getState().increaseFishes()
  })
}
```

How about this?

```

> More details: https://twitter.com/dan_abramov/status/959507572951797761

Reviewer: Not sure how this is helpful. Remove it or replace the link to #302 and add the twitter link there?

Author: OK
